File: 1.ADIM/STEP 1 PYTHON Example/iframe_ornek.py
# Python 3.6+
# Sample codes for STEP 1

import base64
import hmac
import hashlib
import requests
import json

# API Integration Information - You can get it from the INFORMATION page by logging into the store panel.
merchant_id = 'XXXXXX'
merchant_key = b'YYYYYYYYYYYYYY'
merchant_salt = b'ZZZZZZZZZZZZZZ'

# Your customer's email address registered on your site or received via the form
email = 'XXXXXXXX'

# Amount to be collected.
payment_amount = '' # For 9.99, 9.99 * 100 = 999 should be sent.

# Order number: Must be unique for every transaction!! This information is sent back in the notification to your notification page.
merchant_oid = ''

# Your customer's name and surname information registered on your site or obtained through the form
user_name = ''

# Your customer's address information registered on your site or received through the form
user_address = ''

# Your customer's phone number registered on your site or received via the form
user_phone = ''

# The page your customer will be directed to after the payment waiting page
# !!! This page is not the page where you will confirm the order! This is the page where you will only inform your customer!
# !!! The page where we will confirm the order is the "Notification URL" page (See: STEP 2 Folder).
merchant_pending_url = 'http://www.siteniz.com/odeme_basarili.php'

# Customer's cart/order content
user_basket = ''

# EXAMPLE Creating $user_basket - You can multiply arrays according to the number of products
"""
user_basket = base64.b64encode(json.dumps([['Example product 1', '18.00', 1],
               ['Example product 2', '33.25', 2],
               ['Example product 3', '45.42', 1]]).encode())
"""

# !!! If you are running this sample code on your local machine, not on the server
# You should write your external IP address (https://www.whatismyip.com/) here. Otherwise you will get an invalid coinpays_token error.
user_ip = ''

# Here you can choose which language you would like to display your payment page in. Below are sample values. To access the last updated values
# Visit (https://app.coinpays.io/shared/languages)
lang = "tr" #tr-en-de-fr-es-kr-jp-ar-ru-cn-id-ua

# It can be sent as 1 for testing when the store is in live mode.
test_mode = '1'

# Here you can choose in which currency you would like to display your cart. Below are sample values. To access the last updated values
# Visit (https://app.coinpays.io/shared/currencies)
#USD-EUR-TRY-GBP-RUB-CNY-KRW
currency = 'TRY'

# You do not need to make any changes in this section.
hash_str = merchant_id + user_ip + merchant_oid + email + payment_amount + user_basket
coinpays_token = base64.b64encode(hmac.new(merchant_key, hash_str.encode() + merchant_salt, hashlib.sha256).digest())

params = {
        'merchant_id'=>merchant_id,
		'user_ip'=>user_ip,
		'lang'=>lang,
		'currency' => currency,
		'merchant_oid'=>merchant_oid,
		'email'=>email,
		'payment_amount'=>payment_amount,
		'coinpays_token'=>coinpays_token,
		'user_basket'=>user_basket,
		'user_name'=>user_name,
		'user_address'=>user_address,
		'user_phone'=>user_phone,
		'merchant_pending_url'=>merchant_pending_url,
		'test_mode' => test_mode
}

result = requests.post('https://app.coinpays.io/api/get-token', params)
res = json.loads(result.text)

if res['status'] == 'success':
    print(res['token'])

    """
    context = {
        'token': res['token']
    }
    """
else:
    print(result.text)


    """
    # Ödeme formunun açılması için gereken HTML kodlar / Başlangıç #

    <script src="https://app.coinpays.io/assets/js/iframeResizer.min.js"></script>
    <iframe src="https://app.coinpays.io/payment/{ token }" id="coinpaysiframe" frameborder="0" scrolling="no" style="width: 100%;"></iframe>
    <script>iFrameResize({},'#coinpaysiframe');</script>

    # Ödeme formunun açılması için gereken HTML kodlar / Bitiş #
    """

File: STEP 1/STEP 1 PHP Example/iframe_ornek.php
<!doctype html>
<html lang="tr">
<head>
    <meta charset="UTF-8">
    <title>Example Payment Page</title>
</head>
<body>

<div>
    <h1>Example Payment Page</h1>
    <p>Example codes for first step</p>
</div>
<br><br>

<div style="width: 100%;margin: 0 auto;display: table;">

<?php

## Sample codes for STEP 1 ##

####################### REQUIRED FIELDS #######################
#
## API Integration Information - You can get it from the INFORMATION page by logging into the store panel.
$merchant_id 	= '';
$merchant_key 	= '';
$merchant_salt	= '';
#
## Your customer's email address registered on your site or received via the form
$email = "";
#
## Amount to be collected.
$payment_amount	= ""; //For 9.99, 9.99 * 100 = 999 should be sent.
#
## Order number: Must be unique for every transaction!! This information is sent back in the notification to your notification page.
$merchant_oid = "";
#
## Your customer's name and surname information registered on your site or obtained through the form
$user_name = "";
#
## Your customer's address information registered on your site or received through the form
$user_address = "";
#
## Your customer's phone number registered on your site or received via the form
$user_phone = "";
#
## The page your customer will be directed to after the payment waiting page
## !!! This page is not the page where you will confirm the order! This is the page where you will only inform your customer!
## !!! The page where we will confirm the order is the "Notification URL" page (See: STEP 2 Folder).
$merchant_pending_url = "http://www.siteniz.com/odeme_onaylaniyor.php";
#
## Müşterinin sepet/sipariş içeriği
$user_basket = "";
#
/* EXAMPLE Creating $user_basket - You can multiply arrays according to the number of products
$user_basket = base64_encode(json_encode(array(
    array("Example Product 1", "18.00", 1), // 1. Product (Product Name - per price - quantity )
    array("Example Product 2", "33.25", 2), // 2. Product (Product Name - per price - quantity )
    array("Example Product 3", "45.42", 1)  // 3. Product (Product Name - per price - quantity )
)));
*/
############################################################################################

## Kullanıcının IP adresi
if( isset( $_SERVER["HTTP_CLIENT_IP"] ) ) {
    $ip = $_SERVER["HTTP_CLIENT_IP"];
} elseif( isset( $_SERVER["HTTP_X_FORWARDED_FOR"] ) ) {
    $ip = $_SERVER["HTTP_X_FORWARDED_FOR"];
} else {
    $ip = $_SERVER["REMOTE_ADDR"];
}

## !!! If you are running this sample code on your local machine, not on the server
## You should write your external IP address (https://www.whatismyip.com/) here. Otherwise you will get an invalid coinpays_token error.
$user_ip=$ip;
##

## It can be sent as 1 for testing when the store is in live mode.
$test_mode = 0;

## Here you can choose which language you would like to display your payment page in. Below are sample values. To access the last updated values
## Visit (https://app.coinpays.io/shared/languages)
$lang=  "tr"; //tr-en-de-fr-es-kr-jp-ar-ru-cn-id-ua

## Here you can choose in which currency you would like to display your cart. Below are sample values. To access the last updated values
## Visit (https://app.coinpays.io/shared/currencies)
$currency = "TRY";//USD-EUR-TRY-GBP-RUB-CNY-KRW

####### You do not need to make any changes in this section.. #######
$hash_str = $merchant_id .$user_ip .$merchant_oid .$email .$payment_amount .$user_basket;
$coinpays_token=base64_encode(hash_hmac('sha256',$hash_str.$merchant_salt,$merchant_key,true));
$post_vals=array(
    'merchant_id'=>$merchant_id,
    'user_ip'=>$user_ip,
    'lang'=>$lang,
    'currency' => $currency,
    'merchant_oid'=>$merchant_oid,
    'email'=>$email,
    'payment_amount'=>$payment_amount,
    'coinpays_token'=>$coinpays_token,
    'user_basket'=>$user_basket,
    'user_name'=>$user_name,
    'user_address'=>$user_address,
    'user_phone'=>$user_phone,
    'merchant_pending_url'=>$merchant_pending_url,
    'test_mode' => $test_mode
);

$ch=curl_init();
curl_setopt($ch, CURLOPT_URL, "https://app.coinpays.io/api/get-token");
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_POST, 1) ;
curl_setopt($ch, CURLOPT_POSTFIELDS, $post_vals);
curl_setopt($ch, CURLOPT_FRESH_CONNECT, true);
curl_setopt($ch, CURLOPT_TIMEOUT, 20);

// XXX: ATTENTION: If you receive the "SSL certificate problem: unable to get local issuer certificate" warning on your local machine,
// You can open and try the code below. HOWEVER, it is very important that this code remains closed on your server (your real environment) for security reasons!
//curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);

$result = @curl_exec($ch);

if(curl_errno($ch))
    die("COINPAYS DIRECT API connection error. err:".curl_error($ch));

curl_close($ch);

$result=json_decode($result,1);

if($result['status']=='success'){
    $token=$result['token'];
}else{
    die("COINPAYS DIRECT API failed. reason:".$result['reason']);
}
#########################################################################

?>

    <!-- HTML codes required to open the payment form / Start -->
    <script src="https://app.coinpays.io/assets/js/iframeResizer.min.js"></script>
    <iframe src="https://app.coinpays.io/payment/<?php echo $token;?>" id="coinpaysiframe" frameborder="0" scrolling="no" style="width: 100%;"></iframe>
    <script>iFrameResize({},'#coinpaysiframe');</script>
    <!-- HTML codes required to open the payment form / Finish -->

</div>

<br><br>
</body>
</html>
